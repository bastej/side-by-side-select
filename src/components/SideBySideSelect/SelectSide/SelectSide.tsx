import React, { MouseEventHandler, ReactNode } from 'react';

import styles from './styles.module.scss';

type Props = {
  onClick: MouseEventHandler<HTMLDivElement>;
  isActive?: boolean;
  header: string;
  errorOccurred?: boolean;
  children?: ReactNode;
  testID?: string;
};

const SelectSide = ({
  onClick,
  isActive = false,
  header,
  children,
  errorOccurred = false,
  testID,
}: Props): JSX.Element => (
  <div
    className={`${styles.SideContainer} ${errorOccurred ? styles.Error : ''}`}
    onClick={onClick}
    data-testid={testID}
  >
    <div className={styles.SideHeader} style={isActive ? { fontWeight: 'bold' } : {}} data-testid="side-header">
      {header}
    </div>
    {children}
  </div>
);

export default SelectSide;
