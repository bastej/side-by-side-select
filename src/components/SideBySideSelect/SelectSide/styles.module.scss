.SideContainer {
  flex: 1;
  border: 1px solid #ced4da;
  border-radius: 0.2rem;
}

.SideHeader {
  padding: 5px 8px;
  border-bottom: 1px solid #ced4da;
}

.Error {
  border-color: red;
}
