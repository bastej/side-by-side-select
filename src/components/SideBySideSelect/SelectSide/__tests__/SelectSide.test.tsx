import React from "react";
import { create } from "react-test-renderer";
import { render } from "@testing-library/react";

import SelectSide from "../SelectSide";

jest.mock("../styles.module.scss", () => ({
  SideContainer: "side-container",
  SideHeader: "side-header",
  Error: "error",
}));

describe("<SelectSide />", () => {
  it("renders side", () => {
    const root = create(<SelectSide header="header" onClick={jest.fn()} />);

    expect(root).toMatchSnapshot();
  });

  it("renders side with error class if error occurred", () => {
    const root = render(
      <SelectSide header="header" onClick={jest.fn()} errorOccurred testID="side" />
    );

    const side = root.getByTestId("side");

    expect(side.classList.contains("error")).toBe(true);
  });

  it("renders side with bolder header if the side is active", () => {
    const root = render(<SelectSide header="header" onClick={jest.fn()} isActive />);

    const side = root.getByTestId("side-header");

    expect(side.style.fontWeight).toBe("bold");
  });
});
