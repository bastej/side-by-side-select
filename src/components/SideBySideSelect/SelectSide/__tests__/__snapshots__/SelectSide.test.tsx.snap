// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectSide /> renders side 1`] = `
<div
  className="side-container "
  onClick={[MockFunction]}
>
  <div
    className="side-header"
    data-testid="side-header"
    style={Object {}}
  >
    header
  </div>
</div>
`;
